File: app/src/main/java/com/tkdev/weatherapp/model/ForecastViewHolder.java
package com.tkdev.weatherapp.model;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.tkdev.weatherapp.R;

import java.text.SimpleDateFormat;

public class ForecastViewHolder extends RecyclerView.ViewHolder {

    public TextView forecastTempCurrent;
    public TextView forecastTempMin;
    public TextView forecastTempMax;
    public TextView forecastWeatherDescription;
    public TextView forecastDayOfForecast;
    public TextView forecastHourOfForecast;

    private static final String FORECAST_DAY_PATTERN = "EEE";
    private static final String FORECAST_HOUR_PATTERN = "HH:mm";

    public ForecastViewHolder(@NonNull View itemView) {
        super(itemView);
        this.forecastTempCurrent = itemView.findViewById(R.id.forecast_temp_current);
        this.forecastTempMin = itemView.findViewById(R.id.forecast_temp_min);
        this.forecastTempMax = itemView.findViewById(R.id.forecast_temp_max);
        this.forecastWeatherDescription = itemView.findViewById(R.id.forecast_weather);
        this.forecastDayOfForecast = itemView.findViewById(R.id.forecast_day);
        this.forecastHourOfForecast = itemView.findViewById(R.id.forecast_hour);
    }


    public ForecastViewHolder setForecast(Weather forecast) {

        SimpleDateFormat weatherDay = new SimpleDateFormat(FORECAST_DAY_PATTERN);
        SimpleDateFormat weatherHour = new SimpleDateFormat(FORECAST_HOUR_PATTERN);

        String tempCurrent = forecast.getTemperatureCurrent() + Weather.TEMPERATURE_SUFFIX;
        String tempMin = forecast.getTemperatureMin() + Weather.TEMPERATURE_SUFFIX;
        String tempMax = forecast.getTemperatureMax() + Weather.TEMPERATURE_SUFFIX;

        this.forecastTempCurrent.setText(tempCurrent);
        this.forecastTempMin.setText(tempMin);
        this.forecastTempMax.setText(tempMax);
        this.forecastWeatherDescription.setText(forecast.getWeatherDescription());
        this.forecastDayOfForecast.setText(weatherDay.format(forecast.getDate()));
        this.forecastHourOfForecast.setText(weatherHour.format(forecast.getDate()));

        return this;
    }

}

File: app/src/main/java/com/tkdev/weatherapp/forecast/app/ForecastViewHolder.kt
package com.tkdev.weatherapp.forecast.app

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.tkdev.weatherapp.common.domain.RetrofitCalls.Companion.FORECAST_DAY_PATTERN
import com.tkdev.weatherapp.common.domain.RetrofitCalls.Companion.FORECAST_HOUR_PATTERN
import com.tkdev.weatherapp.common.domain.RetrofitCalls.Companion.datePattern
import com.tkdev.weatherapp.common.domain.RetrofitCalls.Companion.temperaturePrefix
import com.tkdev.weatherapp.common.domain.retrofit_data_source.forecast_dto.List
import kotlinx.android.synthetic.main.list_forecast.view.*

class ForecastViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private var forecastTempCurrent: TextView = itemView.forecast_temp_current
    private var forecastTempMin: TextView = itemView.forecast_temp_min
    private var forecastTempMax: TextView = itemView.forecast_temp_max
    private var forecastWeatherDescription: TextView = itemView.forecast_weather
    private var forecastDayOfForecast: TextView = itemView.forecast_day
    private var forecastHourOfForecast: TextView = itemView.forecast_hour

    fun setForecast(forecast: List, forecastTimezone: Int, prefix: String) {
        forecastTempCurrent.text = temperaturePrefix(forecast.main.temp,prefix)
        forecastTempMin.text = temperaturePrefix(forecast.main.tempMin,prefix)
        forecastTempMax.text = temperaturePrefix(forecast.main.tempMax,prefix)
        forecastWeatherDescription.text = forecast.weather[0].main
        forecastDayOfForecast.text = datePattern(forecast.dt, forecastTimezone, FORECAST_DAY_PATTERN)
        forecastHourOfForecast.text = datePattern(forecast.dt, forecastTimezone, FORECAST_HOUR_PATTERN)
    }
}
